product_docs/docs/pgd/5/admin-tpa/upgrading_major_rolling.mdx

```diff
@@ -0,0 +1,537 @@
+Collecting old cluster control data                         ok
+Collecting new cluster control data                         ok
+Checking LSN of new cluster                                 ok
+Starting new cluster with BDR disabled                      ok
+Connecting to new cluster                                   ok
+Creating replication origin (bdr_bdrdb_rb69_bdr2)           ok
+Advancing replication origin (bdr_bdrdb_rb69_bdr2, 0/1F4... ok
+Creating replication origin (bdr_bdrdb_rb69_bdr1)           ok
+Advancing replication origin (bdr_bdrdb_rb69_bdr1, 0/1E8... ok
+Creating replication slot (bdr_bdrdb_rb69_bdr1)             ok
+Creating replication slot (bdr_bdrdb_rb69)                  ok
+Creating replication slot (bdr_bdrdb_rb69_bdr2)             ok
+Stopping new cluster
+```
+
+### Modify the Postgres service
+
+The Postgres service on the system is configured to start the old version 
+of Postgres (PG15). You need to modify the `postgres.service` file to start the new 
+version (PG16). You can do this using `sed` to replace the old version
+number `15` with `16` throughout the file. 
+
+After you've changed the version number, you can tell the 
+systemd daemon to reload the configuration. On the target node, run:
+
+```bash
+sudo sed -i -e  's/15/16/g' /etc/systemd/system/postgres.service
+sudo systemctl daemon-reload
+```
+
+### Start Postgres
+
+Start the modified Postgres service by running:
+
+```bash
+sudo systemctl start postgres
+```
+
+### Validate the Postgres version
+
+Repeating the first step, check the version of Postgres to confirm that you upgraded kaboom correctly. While still on kaboom, run:
+
+```bash
+sudo -u postgres psql bdrdb
+```
+
+```sql
+SELECT * FROM bdr.group_versions_details;
+```
+
+The output is similar to this for your cluster. Use the output to confirm that kaboom is running the upgraded Postgres version. 
+
+```
+  node_id   | node_name |        postgres_version        | bdr_version 
+------------+-----------+--------------------------------+-------------
+ 3490219809 | kaftan    | 15.5 (Debian 15.5-1EDB.buster) | 5.3.0
+ 2710197610 | kaboom    | 16.1 (Debian 16.1-1EDB.buster) | 5.3.0
+ 2111777360 | kaolin    | 15.5 (Debian 15.5-1EDB.buster) | 5.3.0
+(3 rows)
+```
+
+### Test the upgrade
+
+After the upgrade is complete on kaboom, test your databases thoroughly to ensure the upgrade was successful. Being still connected to kaboom ensures that your tests run specifically against the new version without interference from cluster routing mechanisms.
+
+```bash
+sudo -u postgres psql <database_name>
+```
+
+Next, execute your test suite, including application transactions, queries, and other operations critical to your environment. Pay particular attention to performance metrics and compatibility with the new version. The insights gained from this controlled upgrade can inform your approach, helping to ensure a smoother transition for all nodes.
+
+### Clean up post-upgrade
```

Comment:
Ditto this vacuum step is somewhat scary. I'd check with someone (as I seriously don't know) whether a vacuum while joined to the cluster but not write leader could cause any issues.

Reply:
Do we have an eng contact I can use or do I need to find one for this?

Reply:
Reached out in #eng-pgd and this vacuum step is safe:

"This shouldn’t impact on data consistency as VACUUM isn’t replicated, but depending on the size of the database it may generate a bit of replication lag which shouldn’t take too long to catch up." William Ivanski